From 1959bf2c98676da43b8be15456b6ff69cd18e46e Mon Sep 17 00:00:00 2001
From: elbachir-one <bachiralfa@gmail.com>
Date: Mon, 24 Jun 2024 17:57:03 +0100
Subject: [PATCH] Adding focussame to focus on the same programe

---
 config.def.h |  1 +
 dwm.c        | 32 ++++++++++++++++++++++++++++++++
 2 files changed, 33 insertions(+)

diff --git a/config.def.h b/config.def.h
index 9efa774..674b065 100644
--- a/config.def.h
+++ b/config.def.h
@@ -85,6 +85,7 @@ static const Key keys[] = {
 	{ MODKEY,                       XK_period, focusmon,       {.i = +1 } },
 	{ MODKEY|ShiftMask,             XK_comma,  tagmon,         {.i = -1 } },
 	{ MODKEY|ShiftMask,             XK_period, tagmon,         {.i = +1 } },
+    { MODKEY,                       XK_v,      focussame,      {0} },
 	TAGKEYS(                        XK_1,                      0)
 	TAGKEYS(                        XK_2,                      1)
 	TAGKEYS(                        XK_3,                      2)
diff --git a/dwm.c b/dwm.c
index 67c6b2b..94dc747 100644
--- a/dwm.c
+++ b/dwm.c
@@ -165,6 +165,7 @@ static void drawbar(Monitor *m);
 static void drawbars(void);
 static void enternotify(XEvent *e);
 static void expose(XEvent *e);
+static void focussame(const Arg *arg);
 static void focus(Client *c);
 static void focusin(XEvent *e);
 static void focusmon(const Arg *arg);
@@ -786,6 +787,37 @@ expose(XEvent *e)
 		drawbar(m);
 }
 
+void
+focussame(const Arg *arg) {
+    Client *c;
+    XClassHint ch = { NULL, NULL };
+    char *class_name = NULL;
+
+    if (!selmon->sel)
+        return;
+
+    if (XGetClassHint(dpy, selmon->sel->win, &ch)) {
+        class_name = ch.res_class;
+    }
+
+    if (!class_name)
+        return;
+
+    for (c = selmon->clients; c; c = c->next) {
+        if (c != selmon->sel && c->tags & selmon->tagset[selmon->seltags] && XGetClassHint(dpy, c->win, &ch)) {
+            if (strcmp(class_name, ch.res_class) == 0) {
+                focus(c);
+                restack(selmon);
+                XFree(ch.res_class);
+                XFree(ch.res_name);
+                return;
+            }
+            XFree(ch.res_class);
+            XFree(ch.res_name);
+        }
+    }
+}
+
 void
 focus(Client *c)
 {
-- 
2.45.2

